ProblemGenerator/module_dynamic_backtracking.py

```diff
@@ -0,0 +1,134 @@
+import math
+import random
+import copy
+
+import Board.module_board
+import ProblemGenerator.difficulties as difficulties
+from ProblemGenerator.module_problem_base_class import ProblemFinder
+
+
+def get_identifier(x: int, y: int):
+    """
+    Returns the string which identifies the fields in the sudoku for the setValues dictionary.
+    :param x: x-axis of the sudoku field
+    :param y: y-axis of the sudoku field
+    :return: identifier string
+    """
+    return str(x) + "_" + str(y)
+
+
+def split_identifier(identifier: str):
+    """
+    Splits the identifier and returns it as an array.
+    :param identifier: identifier of the dictionary set_values
+    :return: Array
+    """
+    return [int(i) for i in identifier.split("_")]
+
+
+def random_list(length: int):
+    """
+    Builds and returns a randomized array of values from one to the given length.
+
+    :param length: the range for the array
+    :return: randomized array of values within the given length
+    """
+    sorted_list = list(range(1, length))
+    return random.sample(sorted_list, len(sorted_list))
+
+
+def find_and_remove(arr: [], val: int):
+    """
+    Searches the value in the given array and removes it if found
+    :param arr: The array with values
+    :param val: The value to be removed
+    :return: None
+    """
+    for value in arr:
+        if value == val:
+            arr.remove(val)
+
```

Comment:
In my oppinion it would make sense to add a break statement here to stop the for loop, once you found the value.

Reply:
I agree and added it.